public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -509,6 +515,13 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
     } else if (stepMode === 'exact') {
       adjustedInterval = Math.max(stepInterval * intervalFactor, safeInterval);
     }
+
+    if (adjustedInterval === safeInterval) {
+      this.isBelowSafeInterval = true;
```

Comment:
sorry, but i think instance-variables are not a good approach here. if two queries will run at the same time, the results can get mixed up. (i do not know what the solution should be here, i only know i do not recommend the instance-variables-approach) 😄

Reply:
Yeah i think we'll have to come up with another solution 🙂